### PR TITLE
Finally remove wxWidgets

### PR DESCRIPTION
**Pull request by stenzek**
### Description of Changes

A year later and a ton of work, wx is finally gone! Celebrations!

Obviously merging this has to wait until the debugger is also merged (and maybe guitars).

CI probably needs to have some GCC builds added for coverage.

### Rationale behind Changes

Where do I begin....

### Suggested Testing Steps

Make sure Qt didn't break.

### CODE REVIEW

File: pcsx2/GS/Renderers/Metal/GSDeviceMTL.mm [hunk 1]
```diff
@@ -89,7 +89,7 @@ static constexpr bool IsCommandBufferCompleted(MTLCommandBufferStatus status)
 	, m_dev(nil)
 {
 	m_backref->second = this;
-	m_mipmap = theApp.GetConfigI("mipmap");
+	m_mipmap = GSConfig.HWMipmap;
```
Comment (lightningterror) [inline pcsx2/GS/Renderers/Metal/GSDeviceMTL.mm:92]: Seems that compile fails here
https://github.com/PCSX2/pcsx2/actions/runs/3725357872/jobs/6318116478#step:11:855
maybe needs a cast?

**Comment by TheComputerGuy96** [comment]
Qt didn't break (I was able to launch and play a few games fine)

I had to re-add package mode though to make an Arch package :frog:

**Comment by SupervisedThinking** [comment]
Well well well means everyone has to deal with the Qt6 [cross-compile](https://doc.qt.io/qt-6/qt6-buildsystem.html#cross-compiling-qt) $%&! show on Linux once wxWidgets will be removed, this gonna be fun... 🤔

**Comment by F0bes** [comment]
You make Linux users (including me) look horrible. We are an open community, you are able to suggest ideas to us. Instead of passive aggressive remarks, why not make a contribution.

**Comment by SupervisedThinking** [comment]
> You make Linux users (including me) look horrible. We are an open community, you are able to suggest ideas to us. Instead of passive aggressive remarks, why not make a contribution.
> 
> Edit: Looks like the original comment was edited, my response was a response to that.

Well I've deleted the comment because it's a waste of time. I already asked for a [sane Qt5 support for Linux users](https://github.com/PCSX2/pcsx2/issues/5561), like all other major emulator offer as Citra, Yuzu, RPCS3, Dolphin do or just keep wxWidgets - Cemu works just fine. But nobody cares because Windows is the focused platform. I even created a [PR](https://github.com/PCSX2/pcsx2/pull/5510) in the past to fix Linux builds because nobody cared to test the removal of crucial stuff. If you have a look at the wording then the statement is clear: get rid of wxWidgets at all costs, don't care about anything else. I'm pretty sure wxWidgets could stay at least for 2.0 as legacy build for compat but well... I guess I'm wrong. So go on, remove the stuff nobody needs for Windows 10 anymore and Celebrations and stuff.

**Comment by JordanTheToaster** [comment]
> > You make Linux users (including me) look horrible. We are an open community, you are able to suggest ideas to us. Instead of passive aggressive remarks, why not make a contribution.
> > Edit: Looks like the original comment was edited, my response was a response to that.
> 
> Well I've deleted the comment because it's a waste of time. I already asked for a [sane Qt5 support for Linux users](https://github.com/PCSX2/pcsx2/issues/5561), like all other major emulators offer as Citra, Yuzu, RPCS3, Dolphin do or just keep wxWidgets - Cemu works just fine. But nobody cares because Windows 10/11 is the focused platform. I even created a [PR](https://github.com/PCSX2/pcsx2/pull/5510) in the past to fix Linux builds because nobody cared to test the removal of crucial stuff. If you have a look at the wording then the statement is clear: get rid of wxWidgets at all costs, don't care about anything else. I'm pretty sure wxWidgets could stay at least for 2.0 as legacy build for compat but well... I guess I'm wrong. So go on, remove the stuff nobody needs for Windows 10 anymore and Celebrations and stuff.

Continuing to be passive aggressive isn't helping convince anyone.

**Comment by RedDevilus** [comment]
> > You make Linux users (including me) look horrible. We are an open community, you are able to suggest ideas to us. Instead of passive aggressive remarks, why not make a contribution.
> > Edit: Looks like the original comment was edited, my response was a response to that.
> 
> Well I've deleted the comment because it's a waste of time. I already asked for a [sane Qt5 support for Linux users](https://github.com/PCSX2/pcsx2/issues/5561), like all other major emulators offer as Citra, Yuzu, RPCS3, Dolphin do or just keep wxWidgets - Cemu works just fine. But nobody cares because Windows 10/11 is the focused platform. I even created a [PR](https://github.com/PCSX2/pcsx2/pull/5510) in the past to fix Linux builds because nobody cared to test the removal of crucial stuff. If you have a look at the wording then the statement is clear: get rid of wxWidgets at all costs, don't care about anything else. I'm pretty sure wxWidgets could stay at least for 2.0 as legacy build for compat but well... I guess I'm wrong. So go on, remove the stuff nobody needs for Windows 10 anymore and Celebrations and stuff.

WX has long survived its initial projections of survival, and if you dislike it just stick to the older builds. Just like older OS with outdated specs fit with outdated software. Keeping compatibility for everything alive would require postponing progress or adding hacks back in again. If you want to handle that aspect for us, go ahead.

But don't expect us to do it for you, unless you have some valid reasons for atleast 10% of PCSX2 users. Either way we are all voluntuurs so we don't have to do anything that users suggest, but doesn't mean we don't listen and think about how much ideas help PCSX2. Trying to maintain WX was a nightmare when I tried to improve upon the monstrosity, now I don't have to cringe looking at the Qt variant having cleaned up the old mess. WX also has higher input latencies when I compared it to Qt so yeah, it's not done out of spite against users and perhaps you are just used to it but this isn't Bethesda or other emulators.

I apologise if this isn't the response you wanted but you did make a previous GitHub issue that was rejected.

File: CMakeLists.txt [hunk 2]
```diff
@@ -58,27 +58,7 @@ endif()
...
-		INSTALL(FILES "${CMAKE_SOURCE_DIR}/linux_various/PCSX2-linux.sh"          DESTINATION "${CMAKE_INSTALL_BINDIR}")
-	endif()
-elseif(NOT WIN32)
-	if(NOT DISABLE_PCSX2_WRAPPER)
-		# special case to avoid having linux files in windows
-		INSTALL(FILES "${CMAKE_SOURCE_DIR}/linux_various/PCSX2-linux.sh"          DESTINATION "${CMAKE_SOURCE_DIR}/bin")
-	endif()
+if(NOT WIN32 AND NOT DISABLE_PCSX2_WRAPPER)
+	# special case to avoid having linux files in windows
+	INSTALL(FILES "${CMAKE_SOURCE_DIR}/linux_various/PCSX2-linux.sh"          DESTINATION "${CMAKE_SOURCE_DIR}/bin")
```
Comment (arcum42) [inline CMakeLists.txt:63]: This line tells it to install PCSX2-linux.sh, but this PR also deletes that file.

**Comment by SupervisedThinking** [comment]
Since Qt6 is meant for Windows 10 it's used. That's it. If I point out at least to keep it for legacy purposes: won't do. The project literally merged commits hours ago for the WX build but well: won't do. As I said it's a waste of time.

**Comment by RedPanda4552** [comment]
This sounds more like a disorganized rant about Windows being the end of the world than anything about dependency management. Linux users have the AppImage to take care of all that for them. Packagers can use a Qt 6 package when it is made available to them by their distro's package maintainers. End of story. Locked.

**Comment by refractionpcsx2** [comment]
Also FYI, dolphin has moved to qt 6 and made win 10 1908 the minimum requirement. If you're comparing their previous release, ours also works down to Windows 7

Please don't come in here with your disingenuous arguments

**Comment by stenzek** [comment]
This is ridiculous that we have to lock PRs because some users get mad about our choices. Clearly you haven't worked with the PCSX2 code base - the wx frontend is riddled with bugs, race conditions, overengineering and is very difficult to work with. That's why I tossed it out and started fresh.

If you want to use wx, the code is there. You can fork and maintain it yourself. That's the beauty of open source. But don't expect the maintainers to do it for you, when it's actively preventing us from making improvements (e.g. I can't do combined audio+video dumping until wx is gone). Plus, it hasn't been at feature parity in some time.

**Comment by stenzek** [comment]
Note: I've included #6161 in here for now, so we don't need to resolve the conflicts later. After the debugger is merged, this is good to go.

**Pull request merged**